R/MeasureSurvRCLL.R

```diff
@@ -73,17 +73,53 @@ MeasureSurvRCLL = R6::R6Class("MeasureSurvRCLL",
       event = truth[, 2] == 1
       event_times = truth[event, 1]
       cens_times = truth[!event, 1]
-      distr = prediction$distr
 
-      if (!any(event)) { # all censored
-        # survival at outcome time (survived *at least* this long)
-        out[!event] = diag(as.matrix(distr[!event]$survival(cens_times)))
-      } else if (all(event)) { # all uncensored
-        # pdf at outcome time (survived *this* long)
-        out[event] = diag(as.matrix(distr[event]$pdf(event_times)))
-      } else { # mix
-        out[event] = diag(as.matrix(distr[event]$pdf(event_times)))
-        out[!event] = diag(as.matrix(distr[!event]$survival(cens_times)))
+      # Bypass distr6 construction if underlying distr represented by array
+      if (inherits(prediction$data$distr, "array")) {
+        surv = prediction$data$distr
+        if (length(dim(surv)) == 3) {
```

Comment:
we need to do this as in graf score when we have 3d survival array

Reply:
@bblodfon see comment below, the new functiion I made `cdfpdf` checks if it's an array or not

Reply:
I know, the problem is that following the code's logic, what comes most surely afterwards (at least 1 censored obs in the test set) is `cdf = t(1 - surv)` and `t` with a 3d `surv` array doesn't work (you need `aperm` and differentiate between matrix and 3d array then)